python/graphstorm/eval/eval_func.py

```diff
@@ -192,62 +194,112 @@ def eval_roc_auc(logits,labels):
-    for i in range(labels.shape[1]):
-        # AUC is only defined when there is at least one positive data.
-        if np.sum(labels[:, i] == 1) > 0 and np.sum(labels[:, i] == 0) > 0:
-            is_labeled = labels[:, i] == labels[:, i]
-            rocauc_list.append(roc_auc_score(labels[is_labeled, i],
-                                             predicted_labels[is_labeled, i]))
+    # For binary results, the sklearn roc_auc_score function asks 1D inputs of predictions.
+    # And the label is a 1D tensor. For other cases, the sklearn roc_auc_score function asks
+    # nD inputs of predictions. So here we need to check the predictions' 2nd dim for the binary
+    # conditions.
+    if len(predicted_labels.shape)>1 and predicted_labels.shape[1] == 2 \
+        and len(labels.shape) == 1:
+        # Here use the 2nd dim, assuming it is the probability of 1s.
+        rocauc_list.append(roc_auc_score(labels, predicted_labels[:, 1]))
+    else:
+        # mutiple class and multiple labels cases
+        try:
+            labels=labels_to_one_hot(labels, predicted_labels.shape[1])
+        except IndexError as e:
+            logging.error("Failure found during evaluation of the f1 score metric due to" + \
+                      " reason: %s", str(e))
+            raise
+
+        for i in range(labels.shape[1]):
+            # AUC is only defined when there is at least one positive data.
+            if np.sum(labels[:, i] == 1) > 0 and np.sum(labels[:, i] == 0) > 0:
+                is_labeled = labels[:, i] == labels[:, i]
+                rocauc_list.append(roc_auc_score(labels[is_labeled, i],
+                                                predicted_labels[is_labeled, i]))
 
     if len(rocauc_list) == 0:
         logging.error('No positively labeled data available. Cannot compute ROC-AUC.')
         return 0
 
     return sum(rocauc_list) / len(rocauc_list)
 
-
 def eval_acc(pred, labels):
     """compute evaluation accuracy.
     """
-    if pred.dim() > 1:
-        # if pred has dimension > 1, it has full logits instead of final prediction
-        assert th.is_floating_point(pred), "Logits are expected to be float type"
-        pred = pred.argmax(dim=1)
-    # Check if pred is integer tensor
-    assert(not th.is_floating_point(pred) and not th.is_complex(pred)), "Predictions are " \
-        "expected to be integer type"
-    return th.sum(pred.cpu() == labels.cpu()).item() / len(labels)
+    try:
+        if pred.dim() > 1:
+            # if pred has dimension > 1, it has full logits instead of final prediction
+            assert th.is_floating_point(pred), "Multiple dimension logits are expected to be" + \
+                " float type."
+            pred = pred.argmax(dim=1)
+        # Check if pred is integer tensor
+        assert (not th.is_floating_point(pred) and not th.is_complex(pred)), "1D predictions " + \
+            "are expected to be integer type."
+    except (AssertionError, ValueError):
```

Comment:
Why we want to capture the asserts?

Reply:
The current logics are based on the following situations:
1. As our latest discussion, if anything wrong for the input data of evaluation functions, the code show break and stop.
2. Our test cases should test the invalid cases, including but not limited to wrong dimension, wrong rows, etc.
3. Our test cases should not let the invalid cases break the testing pipeline, i.e., test code should be able to catch errors occurred in the `eval_func.py`.

Based on the three situations, I used the try/catch and raise the assertion errors. Use the current method, these Errors will be raised to function callers. 
- If no one handle these errors, the code will break, just as we expected for users who call our APIs.
- There are test cases in test code to mimic these invalid inputs and can trigger the correct error to happen.
- Our test code catch these errors and handle them properly to let the test pipeline pass through.